barretenberg/cpp/src/barretenberg/transcript/transcript.hpp

```diff
@@ -232,20 +235,20 @@ class BaseTranscript {
 
         // Compute the new challenge buffer from which we derive the challenges.
 
-        // Create challenges from bytes.
+        // Create challenges from Frs.
         std::array<uint256_t, num_challenges> challenges{};
 
         // Generate the challenges by iteratively hashing over the previous challenge.
         for (size_t i = 0; i < num_challenges; i++) {
             auto next_challenge_buffer = get_next_challenge_buffer(); // get next challenge buffer
-            std::array<uint8_t, sizeof(uint256_t)> field_element_buffer{};
+            Fr field_element_buffer = next_challenge_buffer;
             // copy half of the hash to lower 128 bits of challenge
             // Note: because of how read() from buffers to fields works (in field_declarations.hpp),
             // we use the later half of the buffer
-            std::copy_n(next_challenge_buffer.begin(),
-                        HASH_OUTPUT_SIZE / 2,
-                        field_element_buffer.begin() + HASH_OUTPUT_SIZE / 2);
-            challenges[i] = from_buffer<uint256_t>(field_element_buffer);
+            // std::copy_n(next_challenge_buffer.begin(),
+            //             HASH_OUTPUT_SIZE / 2,
+            //             field_element_buffer.begin() + HASH_OUTPUT_SIZE / 2);
+            challenges[i] = uint256_t(field_element_buffer);
```

Comment:
Using uint256_t here because it has the exact conversion function we want from bn254 fr to grumpkin fr, but the stdlib version should not actually just do this.